pkg/apis/databases/v1alpha4/vault_connection.go

```diff
@@ -142,7 +142,7 @@ func (d *Database) getVaultConnection(ctx context.Context, clientset kubernetes.
 		return "", "", errors.Wrap(err, "failed to unmarshal response")
 	}
 
-	uriTemplate, err := getConnectionURITemplate(valueOrValueFrom.ValueFrom.Vault, loginResponse.Auth.ClientToken, d.Name)
```

Comment:
This doesn't seem right to me. The `getConnectionURITemplate` function returns two values, the second is an `err`. Does this compile?